### PR TITLE
Migrate account_product_fiscal_classification to V10

### PR DESCRIPTION
**Pull request by yk2kus**

### CODE REVIEW

**Comment by yk2kus** [comment]
I need help with the syntax error
https://goo.gl/QZypwc

**Comment by rvalyi** [comment]
@yk2kus I'm not very sure, but for the rst error, you can try to remove 2 spaces at lines 22 and 23 of the README.rst file. If it doesn't do the trick, you can also try to insert a blank line before line 22, but only if removing space alone doesn't fix the syntax problem.

**Comment by renatonlima** [comment]
:+1:

File: account_product_fiscal_classification/models/wizard_change_fiscal_classification.py
Comment (rvalyi) [inline account_product_fiscal_classification/models/wizard_change_fiscal_classification.py:30]: :-1: sorry if I'm wrong but shouldn't we have an @api.multi decorator here then?

Comment (yk2kus) [reply account_product_fiscal_classification/models/wizard_change_fiscal_classification.py:30]: @rvalyi not required but can be placed. It will not have any impact in the result because old api is not supported in V10, also will not raise any issue because we can loop through a single recordset as well. I would like to know if it will make an impact ?

Comment (rvalyi) [reply account_product_fiscal_classification/models/wizard_change_fiscal_classification.py:30]: In fact, I think api.one is not deprecated anymore, look at all the compute methods in account/models/account_invoice.py for instance, they all use the @api.one formalism. Can we not switch back to @api.one and the former impl assuming self is the record?
see for instance https://github.com/odoo/odoo/blob/10.0/addons/account/models/account_invoice.py#L88

**Comment by yk2kus** [comment]
From what I can understand, it is deprecated but not discontinued. Source https://github.com/odoo/odoo/issues/12408#issuecomment-226421501 . My purpose of removal was lint check https://goo.gl/CVVnks

File: account_product_fiscal_classification/models/wizard_change_fiscal_classification.py
Comment (yk2kus) [reply account_product_fiscal_classification/models/wizard_change_fiscal_classification.py:30]: From what I can understand, it is deprecated but not discontinued. Source https://github.com/odoo/odoo/issues/12408#issuecomment-226421501 . My purpose of removal was lint check https://goo.gl/CVVnks , should we still switch back?

Comment (yk2kus) [reply account_product_fiscal_classification/models/wizard_change_fiscal_classification.py:30]: @rvalyi waiting for your response.

Comment (lasley) [reply account_product_fiscal_classification/models/wizard_change_fiscal_classification.py:30]: @yk2kus - @rvalyi is correct that `api.one` is deprecated and should not be used in new code.

You can either alter to `api.multi` or switch to [oca_decorators](https://github.com/OCA/oca-decorators), which has a [`foreach`](https://github.com/OCA/oca-decorators/blob/master/oca/decorators/foreach.py#L9) to replace the deprecated `.one`

File: account_product_fiscal_classification/demo/product_category.xml
Comment (lasley) [inline account_product_fiscal_classification/demo/product_category.xml:8]: Please remove the `data` tags - they are implied with the new `odoo` tag

File: account_product_fiscal_classification/models/account_product_fiscal_classification.py
Comment (lasley) [inline account_product_fiscal_classification/models/account_product_fiscal_classification.py:52]: Use `.with_context(active_test=False)` instead of the `|` on `active`

File: account_product_fiscal_classification/models/account_product_fiscal_classification_model.py
Comment (lasley) [inline account_product_fiscal_classification/models/account_product_fiscal_classification_model.py:21]: Please switch long headers to [shortened header style](https://github.com/OCA/maintainer-tools/blob/master/template/module/__openerp__.py#L3)

File: account_product_fiscal_classification/models/product_template.py
Comment (lasley) [inline account_product_fiscal_classification/models/product_template.py:56]: `''.join(template.categ_id.fiscal_classification_ids.mapped('name'))`

**Review by lasley — CHANGES_REQUESTED** [review]
Thanks @yk2kus - code, no test

**Comment by rvalyi** [comment]
guys, I'm closing this PR as it has been superseded by https://github.com/OCA/account-fiscal-rule/pull/74 which has been merged.

**Pull request closed without merging**
